### ISSUE TITLE
`HasErrors` property changed must be raised when errors changed

### ISSUE THREAD

**Issue by xperiandri** [elmish/Elmish.WPF#603]
I tried to implement that myself but I don't understand how to determine if errors changed.
You mutate that Map constantly

**Comment by xperiandri** [comment]
What I want is to disable the primary dialog button when my model is not valid which is the initial state of my model

```
<ContentDialog x:Class="Construction.Dialogs.CreateBuildingDialog"
               xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
               xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
               xmlns:local="using:eCierge.Console.Construction.Dialogs"
               xmlns:p="using:eCierge.Console.Construction.Presentation"
               xmlns:ctb="using:CommunityToolkit.WinUI.Behaviors"
               xmlns:i="using:Microsoft.Xaml.Interactivity"
               xmlns:ic="using:Microsoft.Xaml.Interactions.Core"

               xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
               xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
               mc:Ignorable="d"

               Title="Create building"
               Style="{StaticResource ScrollableContentDialogStyle}"
               MinWidth="400"
               MinHeight="300"
               PrimaryButtonText="Create"
               PrimaryButtonCommand="{x:Bind ViewModel.CreateBuildingCommand, Mode=OneWay}"
               IsPrimaryButtonEnabled="{x:Bind ViewModel.HasErrors, Converter={StaticResource BoolNegationConverter}, Mode=OneWay}"
               SecondaryButtonText="Cancel"
               SecondaryButtonCommand="{x:Bind ViewModel.CancelCommand, Mode=OneWay}">

  <i:Interaction.Behaviors>
    <ic:EventTriggerBehavior EventName="{x:Bind CloseDialogEventName}" SourceObject="{x:Bind DataContext, Mode=OneWay}">
      <ic:CallMethodAction MethodName="{x:Bind HideMethodName}" TargetObject="{x:Bind}" />
    </ic:EventTriggerBehavior>
  </i:Interaction.Behaviors>

  <p:CreateBuildingPage MinWidth="400" />
</ContentDialog>
```

**Comment by xperiandri** [comment]
The fix is in my for now

**Comment by marner2** [comment]
I see, it's definitely an oversight. Creating tests and updates now...

**Comment by marner2** [comment]
Should be fixed with [4.0.0-beta-56 on NuGet](https://www.nuget.org/packages/Elmish.WPF/4.0.0-beta-56) when that goes live.
